src/main/java/reactor/core/publisher/FluxMapOrFilterFuseable.java

```diff
@@ -0,0 +1,450 @@
+		super(source);
+		if (!(source instanceof Fuseable)) {
+			throw new IllegalArgumentException("The source must implement the Fuseable interface for this operator to work");
+		}
+		this.mapper = Objects.requireNonNull(mapper, "mapper");
+	}
+
+	public Function<? super T, ? extends R> mapper() {
+		return mapper;
+	}
+
+	@Override
+	public void subscribe(Subscriber<? super R> s) {
+		if (s instanceof ConditionalSubscriber) {
+			
+			ConditionalSubscriber<? super R> cs = (ConditionalSubscriber<? super R>) s;
+			source.subscribe(new MapOrFilterFuseableConditionalSubscriber<>(cs, mapper));
+			return;
+		}
+		source.subscribe(new MapOrFilterFuseableSubscriber<>(s, mapper));
+	}
+
+	static final class MapOrFilterFuseableSubscriber<T, R>
+			implements Subscriber<T>, Receiver, Producer, Loopback, Subscription,
+			           SynchronousSubscription<R>, Trackable {
+		final Subscriber<? super R>			actual;
+		final Function<? super T, ? extends R> mapper;
+
+		boolean done;
+
+		QueueSubscription<T> s;
+
+		int sourceMode;
+
+		public MapOrFilterFuseableSubscriber(Subscriber<? super R> actual, Function<? super T, ? extends R> mapper) {
+			this.actual = actual;
+			this.mapper = mapper;
+		}
+
+		@SuppressWarnings("unchecked")
+		@Override
+		public void onSubscribe(Subscription s) {
+			if (Operators.validate(this.s, s)) {
+				this.s = (QueueSubscription<T>)s;
+				actual.onSubscribe(this);
+			}
+		}
+
+		@Override
+		public void onNext(T t) {
+			if (done) {
+				Exceptions.onNextDropped(t);
+				return;
+			}
+
+			int m = sourceMode;
+			
+			if (m == NONE) {
+				R v;
+	
+				try {
+					v = mapper.apply(t);
+				} catch (Throwable e) {
+					onError(Exceptions.mapOperatorError(s, e, t));
+					return;
+				}
+	
+				if (v == null) {
+					s.request(1);
+				}
+				else{
+					actual.onNext(v);
+				}
+	
+			} else
+			if (m == ASYNC) {
+				actual.onNext(null);
+			}
+		}
+
+		@Override
+		public void onError(Throwable t) {
+			if (done) {
+				Exceptions.onErrorDropped(t);
+				return;
+			}
+
+			done = true;
+
+			actual.onError(t);
+		}
+
+		@Override
+		public void onComplete() {
+			if (done) {
+				return;
+			}
+			done = true;
+
+			actual.onComplete();
+		}
+
+		@Override
+		public boolean isStarted() {
+			return s != null && !done;
+		}
+
+		@Override
+		public boolean isTerminated() {
+			return done;
+		}
+
+		@Override
+		public Object downstream() {
+			return actual;
+		}
+
+		@Override
+		public Object connectedInput() {
+			return mapper;
+		}
+
+		@Override
+		public Object upstream() {
+			return s;
+		}
+		
+		@Override
+		public void request(long n) {
+			s.request(n);
+		}
+		
+		@Override
+		public void cancel() {
+			s.cancel();
+		}
+
+		@Override
+		public R poll() {
+			if (sourceMode == ASYNC) {
+				long dropped = 0;
+				for (;;) {
+					T v = s.poll();
+					if(v != null) {
+						R u = mapper.apply(v);
+						if (u == null) {
+							if (dropped != 0) {
```

Comment:
This should also happen on the v == null path because otherwise all the dropped elements won't get replenished (I'll fix `filter`)